std/algorithm/iteration.d

```diff
@@ -422,7 +422,10 @@ private struct _Cache(R, bool bidir)
             auto opSlice(size_t low, size_t high)
             in
             {
-                assert(low <= high);
+                assert(
+                    low <= high,
+                    "Attempting to slice a cache with a larger first argument than the second"
+                );
```

Comment:
Four lines is really overdoing it. Also the message is prolix.

```d
assert(low <= high, "Bounds error in cache slicing.");
```

Also you need to decide whether you always end with the period or not (you did so far).